diff --git a/libiberty/setenv.c b/libiberty/setenv.c
index ba5abfc..17d9e31 100644
--- a/libiberty/setenv.c
+++ b/libiberty/setenv.c
@@ -81,6 +81,13 @@ static char **last_environ;
 int
 setenv (const char *name, const char *value, int replace)
 {
+#ifdef _WIN32
+  if (!replace) {
+    if (getenv(name))
+      return 0;
+  }
+  return _putenv_s(name, value ? value : "");
+#else
   register char **ep = 0;
   register size_t size;
   const size_t namelen = strlen (name);
@@ -157,6 +164,7 @@ setenv (const char *name, const char *value, int replace)
   UNLOCK;
 
   return 0;
+#endif
 }
 
 void
