diff --git a/libcxx/src/filesystem/file_descriptor.h b/libcxx/src/filesystem/file_descriptor.h
index d3a668f..41017cf 100644
--- a/libcxx/src/filesystem/file_descriptor.h
+++ b/libcxx/src/filesystem/file_descriptor.h
@@ -54,8 +54,10 @@ file_type get_file_type(DirEntT* ent, int) {
     return file_type::symlink;
   case DT_REG:
     return file_type::regular;
+#if defined(DT_SOCK)
   case DT_SOCK:
     return file_type::socket;
+#endif
   // Unlike in lstat, hitting "unknown" here simply means that the underlying
   // filesystem doesn't support d_type. Report is as 'none' so we correctly
   // set the cache to empty.
