diff --git a/compiler-rt/lib/scudo/standalone/memtag.h b/compiler-rt/lib/scudo/standalone/memtag.h
index c48e228..7a0cfe4 100644
--- a/compiler-rt/lib/scudo/standalone/memtag.h
+++ b/compiler-rt/lib/scudo/standalone/memtag.h
@@ -62,6 +62,9 @@ inline uint8_t extractTag(uptr Ptr) {
 inline bool systemSupportsMemoryTagging() {
 #ifndef HWCAP2_MTE
 #define HWCAP2_MTE (1 << 18)
+#endif
+#ifndef AT_HWCAP2
+#define AT_HWCAP2 26
 #endif
   return getauxval(AT_HWCAP2) & HWCAP2_MTE;
 }
