From fc5a1b72617ba1e180a41c2a22043ab9cbd9884f Mon Sep 17 00:00:00 2001
From: Shengyun Zhou <GGGZ-1101-28@live.cn>
Date: Mon, 19 Sep 2022 11:45:46 +0800
Subject: [PATCH 2/3] wamrc: link static libssp for MinGW

---
 wamr-compiler/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wamr-compiler/CMakeLists.txt b/wamr-compiler/CMakeLists.txt
index 15acdd49..7dc98c06 100644
--- a/wamr-compiler/CMakeLists.txt
+++ b/wamr-compiler/CMakeLists.txt
@@ -276,7 +276,7 @@ if (NOT MSVC)
   target_link_libraries (wamrc aotclib vmlib ${WAMRC_LINK_LLVM_LIBS} ${lib_ubsan}
                          -lm -lpthread ${lib_lldb} ${UV_A_LIBS})
   if (MINGW)
-      target_link_libraries (wamrc -lssp -lws2_32)
+      target_link_libraries (wamrc ssp.a ws2_32)
   else()
       target_link_libraries (wamrc -ldl)
   endif()
-- 
2.37.3

