From 3e4b9110d6a888d62b0828491b5bc936fc199734 Mon Sep 17 00:00:00 2001
From: Shengyun Zhou <GGGZ-1101-28@live.cn>
Date: Tue, 20 Sep 2022 16:30:45 +0800
Subject: [PATCH 3/3] wamrc: support installing wamrc

---
 wamr-compiler/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/wamr-compiler/CMakeLists.txt b/wamr-compiler/CMakeLists.txt
index 7dc98c06..70ef97b5 100644
--- a/wamr-compiler/CMakeLists.txt
+++ b/wamr-compiler/CMakeLists.txt
@@ -284,3 +284,5 @@ else()
   target_link_libraries (wamrc aotclib vmlib  ${lib_lldb} ${WAMRC_LINK_LLVM_LIBS} ${lib_ubsan}
                          ${UV_A_LIBS})
 endif()
+
+install (TARGETS wamrc DESTINATION bin)
-- 
2.37.3

