From 913775a520e59132730a3ae2fe700332ed9a3852 Mon Sep 17 00:00:00 2001
From: Shengyun Zhou <GGGZ-1101-28@live.cn>
Date: Thu, 13 Oct 2022 10:19:34 +0800
Subject: [PATCH] wamrc: fix compilation error for MinGW GCC

---
 wamr-compiler/CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/wamr-compiler/CMakeLists.txt b/wamr-compiler/CMakeLists.txt
index e4113d05..64bd33ea 100644
--- a/wamr-compiler/CMakeLists.txt
+++ b/wamr-compiler/CMakeLists.txt
@@ -230,10 +230,6 @@ endif ()
 if (NOT MSVC)
   set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fstack-protector-strong --param ssp-buffer-size=4")
 endif()
-if (NOT (MSVC OR CMAKE_C_COMPILER MATCHES ".*clang.*" OR CMAKE_C_COMPILER_ID MATCHES ".*Clang"))
-  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wl,-z,noexecstack,-z,relro,-z,now")
-  set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wl,-z,noexecstack,-z,relro,-z,now")
-endif()
 
 # We disable these flags by default to stay the same with wasm runtime
 # set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mindirect-branch=thunk -mfunction-return=thunk")
-- 
2.38.0

