# Source: https://github.com/loongson-community/musl
diff --git a/arch/loongarch64/kstat.h b/arch/loongarch64/kstat.h
new file mode 100644
index 00000000..f66ed2e9
--- /dev/null
+++ b/arch/loongarch64/kstat.h
@@ -0,0 +1,22 @@
+struct kstat {
+    unsigned long   st_dev;     /* Device.  */
+    unsigned long   st_ino;     /* File serial number.  */
+    unsigned int    st_mode;    /* File mode.  */
+    unsigned int    st_nlink;   /* Link count.  */
+    unsigned int    st_uid;     /* User ID of the file's owner.  */
+    unsigned int    st_gid;     /* Group ID of the file's group. */
+    unsigned long   st_rdev;    /* Device number, if device.  */
+    unsigned long   __pad1;
+    long        st_size;    /* Size of file, in bytes.  */
+    int     st_blksize; /* Optimal block size for I/O.  */
+    int     __pad2;
+    long        st_blocks;  /* Number 512-byte blocks allocated. */
+    long        st_atime_sec;   /* Time of last access.  */
+    unsigned long   st_atime_nsec;
+    long        st_mtime_sec;   /* Time of last modification.  */
+    unsigned long   st_mtime_nsec;
+    long        st_ctime_sec;   /* Time of last status change.  */
+    unsigned long   st_ctime_nsec;
+    unsigned int    __unused4;
+    unsigned int    __unused5;
+};
