diff --git a/libc-top-half/musl/src/network/lookup_name.c b/libc-top-half/musl/src/network/lookup_name.c
index aa558c1..bb372a6 100644
--- a/libc-top-half/musl/src/network/lookup_name.c
+++ b/libc-top-half/musl/src/network/lookup_name.c
@@ -13,7 +13,6 @@
 #include <resolv.h>
 #include "lookup.h"
 #include "stdio_impl.h"
-#include "syscall.h"
 
 static int is_valid_hostname(const char *host)
 {
diff --git a/libc-top-half/musl/src/network/res_msend.c b/libc-top-half/musl/src/network/res_msend.c
index 3e01800..de2815d 100644
--- a/libc-top-half/musl/src/network/res_msend.c
+++ b/libc-top-half/musl/src/network/res_msend.c
@@ -11,12 +11,11 @@
 #include <errno.h>
 #include <pthread.h>
 #include "stdio_impl.h"
-#include "syscall.h"
 #include "lookup.h"
 
 static void cleanup(void *p)
 {
-	__syscall(SYS_close, (intptr_t)p);
+	close((intptr_t)p);
 }
 
 static unsigned long mtime()
