diff --git a/Makefile b/Makefile
index e8cc443..6e11f11 100644
--- a/Makefile
+++ b/Makefile
@@ -159,12 +159,14 @@ obj/%.lo: $(srcdir)/%.c $(GENH) $(IMPH)
 	$(CC_CMD)
 
 lib/libc.so: $(LOBJS) $(LDSO_OBJS)
+	echo "$(LOBJS)" "$(LDSO_OBJS)" "$(LIBCC)" | sed 's/\\/\//g' > libc.so.rsp
 	$(CC) $(CFLAGS_ALL) $(LDFLAGS_ALL) -nostdlib -shared \
-	-Wl,-e,_dlstart -o $@ $(LOBJS) $(LDSO_OBJS) $(LIBCC)
+	-Wl,-e,_dlstart -o $@ @libc.so.rsp
 
 lib/libc.a: $(AOBJS)
 	rm -f $@
-	$(AR) rc $@ $(AOBJS)
+	echo "$(AOBJS)" > libc.a.rsp
+	$(AR) rc $@ @libc.a.rsp
 	$(RANLIB) $@
 
 $(EMPTY_LIBS):
